extensions/Worker.Extensions.CosmosDB/src/Config/CosmosDBBindingOptions.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using Azure.Core;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Functions.Worker.Extensions.CosmosDB;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    internal class CosmosDBBindingOptions
+    {
+        public string? ConnectionString { get; set; }
+
+        public string? AccountEndpoint { get; set; }
+
+        public TokenCredential? Credential { get; set; }
+
+        internal CosmosClient? Client { get; set; }
+
+        internal virtual CosmosClient GetClient(string preferredLocations = "")
```

Comment:
I don't think a singular client is correct due to `preferredLocations`. We may still need the dictionary cache for these different clients.

Reply:
Ahh fair point

Reply:
@jviau updated again xD let me know what you think